### PR TITLE
[Feat] add 'children' prop to Resources

### PR DESCRIPTION
**Pull request by hiaselhans**
React admin makes it fast to develop admin-interfaces with react.
However, when resources get more and more functions it is a bit unhandy to pass CustomRoutes to <Admin/>

This PR adds children as a property of resources and makes it possible to define the whole resource with custom sub-routes in a seperate typescript module:

``` tsx
import ListView from "./list";
import CustomRoute from "./customroute";

const MyResource = () => {
    return {
        list: ListView,
        children: <CustomRoute/>
    }
}
```

### CODE REVIEW

**Comment by fzaninotto** [comment]
Thanks! Thats an interesting feature. 

As it's an enhancement, please rebase your PR against the `next` branch. 

Also, please document that feature and add some unit tests to make sure it works.

File: docs/CustomRoutes.md [hunk 1]
```diff
@@ -5,21 +5,24 @@ title: "The CustomRoutes Component"
...
+To register your own routes, pass one or several `<CustomRoutes>` elements as children of `<Admin>`. Declare as many [react-router-dom](https://reactrouter.com/docs/en/v6/api#routes-and-route) `<Route>` as you want inside them.
+Alternatively, you can add your custom routes to resources. They will be available under the resource prefix.
 
 ```jsx
 // in src/App.js
 import * as React from "react";
 import { Admin, Resource, CustomRoutes } from 'react-admin';
 import { Route } from "react-router-dom";
-import posts from './posts';
+import posts, {PostAnalytics} from './posts';
```
Comment (WiXSL) [inline docs/CustomRoutes.md:16]:
```suggestion
import posts, { PostAnalytics } from './posts';
```

File: docs/CustomRoutes.md [hunk 2]
```diff
@@ -5,21 +5,24 @@ title: "The CustomRoutes Component"
...
 import comments from './comments';
 import Settings from './Settings';
 import Profile from './Profile';
 
 const App = () => (
     <Admin dataProvider={simpleRestProvider('http://path.to.my.api')}>
-        <Resource name="posts" {...posts} />
+        <Resource name="posts" {...posts}>
+            <Route path="analytics" element={<PostAnalytics/>}>
```
Comment (WiXSL) [inline docs/CustomRoutes.md:24]:
```suggestion
            <Route path="analytics" element={<PostAnalytics/>} />
```

File: docs/CustomRoutes.md [hunk 3]
```diff
@@ -5,21 +5,24 @@ title: "The CustomRoutes Component"
...
 
-To register your own routes, pass one or several `<CustomRoutes>` elements as children of `<Admin>`. Declare as many [react-router-dom](https://reactrouter.com/docs/en/v6/api#routes-and-route) `<Route>` as you want inside them:
+To register your own routes, pass one or several `<CustomRoutes>` elements as children of `<Admin>`. Declare as many [react-router-dom](https://reactrouter.com/docs/en/v6/api#routes-and-route) `<Route>` as you want inside them.
+Alternatively, you can add your custom routes to resources. They will be available under the resource prefix.
 
 ```jsx
 // in src/App.js
 import * as React from "react";
 import { Admin, Resource, CustomRoutes } from 'react-admin';
 import { Route } from "react-router-dom";
```
Comment (WiXSL) [inline docs/CustomRoutes.md:15]:
```suggestion
import { Route } from "react-router-dom";
import simpleRestProvider from 'ra-data-simple-rest';
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

**Review by WiXSL — CHANGES_REQUESTED** [review]
Thanks for your work!
You are still missing some test to prove that it works.

File: docs/CustomRoutes.md (hunk 3 above)
Comment (hiaselhans) [reply docs/CustomRoutes.md:15]: good catch!

**Comment by hiaselhans** [comment]
thx @WiXSL !

i fixed the comments and added a test.
In my local env 3 unrelated tests are failing. However rebasing to next will flood the changeset in this PR

**Review by fzaninotto — APPROVED** [review]

**Comment by fzaninotto** [comment]
Thanks!

**Comment by hiaselhans** [comment]
Great, thank you!

**Pull request merged**